app/src/main/java/com/github/leuludyha/ibdb/presentation/screen/openLibrary/OpenLibraryViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package com.github.leuludyha.ibdb.presentation.screen.openLibrary
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+import dagger.hilt.android.lifecycle.HiltViewModel
+
+import com.github.leuludyha.domain.model.Search
+import com.github.leuludyha.domain.useCase.SearchUseCase
+import com.github.leuludyha.domain.util.Result
+
+
+@HiltViewModel
+class OpenLibraryViewModel @Inject constructor(
```

Comment:
Again I don't think a ViewModel has anything to do with the OpenLibrary API.

Reply:
I could have chosen a better name but `OpenLibrary` here doesn't refer to the website but to a possible `Screen` that we would create later. However we probably wouldn't name it `OpenLibrary`. But in any case a `ViewModel` is inherently linked to a specific `View` (`Screen` in our architecture)